at_secondary/at_secondary_server/lib/src/verb/handler/scan_verb_handler.dart

```diff
@@ -107,42 +111,60 @@ class ScanVerbHandler extends AbstractVerbHandler {
+  ///
+  /// **Parameters**
+  ///
+  /// [atConnectionMetadata] Metadata of the inbound connection.
+  ///
+  /// [keys] List of keys from the secondary persistent store.
+  ///
+  /// **Returns**
+  ///
+  /// Returns the list of keys of current atsign.
+  List<String> _getLocalKeys(
+      InboundConnectionMetadata atConnectionMetadata, List<String> keys) {
+    List<String> keysList = [];
     // Verify if the current user is authenticated or not
     // If authenticated get all the keys except for private keys
     // If not, get only public keys
     if (atConnectionMetadata.isAuthenticated) {
       //display all keys except private
-      keys.removeWhere((key) =>
-          key.toString().startsWith('privatekey:') ||
-          key.toString().startsWith('public:_') ||
-          key.toString().startsWith('private:'));
-      keyString = keys.toString();
+      keys.removeWhere((key) => _isPrivateKeyForAtSign(key));
+      keysList = keys;
     } else {
-      //When pol is performed, display keys that are private to the atsign.
+      // When pol is performed, display keys that are private to the atsign.
       if (atConnectionMetadata.isPolAuthenticated) {
-        keys.removeWhere((test) =>
-            (test
-                    .toString()
-                    .startsWith('${atConnectionMetadata.fromAtSign}:') ==
-                false) ||
-            test.toString().startsWith('public:_'));
-        keyString = keys
-            .toString()
-            .replaceAll('${atConnectionMetadata.fromAtSign}:', '');
+        keys.removeWhere((key) => _isPrivateKeyForAtSign(key,
+            fromAtSign: atConnectionMetadata.fromAtSign));
+        // Remove the atSigns from the inbound connection
+        // keys and add the modified key to the list.
+        // @murali:phone@sitaram => phone@sitaram
+        for (var key in keys) {
+          var modifiedKey =
+              key.replaceAll('${atConnectionMetadata.fromAtSign}:', '');
+          keysList.add(modifiedKey);
+        }
       } else {
         // When pol is not performed, display only public keys
-        keys.removeWhere((test) =>
-            test.toString().startsWith('public:_') ||
-            !test.toString().startsWith('public:'));
-        keyString = keys.toString().replaceAll('public:', '');
+        keys.removeWhere((key) => _isPrivateKeyForAtSign(key));
+        for (var key in keys) {
+          var modifiedKey = key.toString().replaceAll('public:', '');
+          keysList.add(modifiedKey);
+        }
       }
     }
-    return keyString;
+    return keysList;
+  }
+
+  /// Checks if a key starts with `public:`, `public:_`, `private:`, `privatekey:` given.
+  /// [key] : The key to check.
+  /// [fromAtSign] : From atsign of the key.
+  /// Returns true if key starts with pattern, else false.
+  bool _isPrivateKeyForAtSign(String key, {String? fromAtSign}) {
+    return key.toString().startsWith('$fromAtSign:') ||
+        key.toString().startsWith('public:') ||
```

Comment:
you have to remove this condition
key.toString().startsWith('public:')

Reply:
I didn't get what exactly you are talking

Reply:
In the current trunk the code reads
```
keys.removeWhere((key) =>
          key.toString().startsWith('privatekey:') ||
          key.toString().startsWith('public:_') ||
          key.toString().startsWith('private:'));
```

But in your new _isPrivateKeyForAtSign you have
```
bool _isPrivateKeyForAtSign(String key, {String? fromAtSign}) {
    return key.toString().startsWith('$fromAtSign:') ||
        key.toString().startsWith('public:') ||
        key.toString().startsWith('private:') ||
        key.toString().startsWith('privatekey:') ||
        key.toString().startsWith('public:_');
  }
```

i.e. you've changed the logic erronesouly by adding `key.toString().startsWith('public:')`

Reply:
Yes got it gary. I was testing the changes but it was being failed. Need to work on it a bit more. I spend my time on this with a calm mind.
Thank you.